### PR TITLE
Allow Aggregation Fuzzer to generate random complex types

### PR DESCRIPTION
**Pull request by mbasmanova**
Fix bugs in min_by, max_by, set_agg, multimap_agg, and arbitrary discovered by expanded testing.

Discovered bugs in min/max #6314, sum/max_data_size_for_stats #6330, and avg_merge_xxx #6344

Discovered bugs in contains, subscript, array_position #6346

### CODE REVIEW

File: velox/vector/fuzzer/VectorFuzzer.h [hunk 1]
```diff
@@ -337,4 +337,12 @@ class VectorFuzzer {
   FuzzerGenerator rng_;
 };
 
+// Generates a random type, including maps, vectors, and arrays. maxDepth
```
Comment (xiaoxmeng) [inline velox/vector/fuzzer/VectorFuzzer.h:340]: nit: /// for comments in header? thanks!

**Review by xiaoxmeng — APPROVED** [review]
@mbasmanova LGTM. Thanks!

File: velox/expression/tests/ArgumentTypeFuzzer.cpp [hunk 2]
```diff
@@ -72,10 +48,14 @@ void ArgumentTypeFuzzer::determineUnboundedTypeVariables() {
     // Random randomType() never generates unknown here.
     // TODO: we should extend randomType types and exclude unknown based
     // on variableInfo.
-    bindings_[variableName] = randomType(rng_);
+    bindings_[variableName] = randType();
   }
 }
 
+TypePtr ArgumentTypeFuzzer::randType() {
+  return velox::randType(rng_, 2);
```
Comment (bikramSingh91) [inline velox/expression/tests/ArgumentTypeFuzzer.cpp:56]: wondering if we would get any value from increasing the depth here?

File: velox/functions/prestosql/aggregates/ArbitraryAggregate.cpp [hunk 3]
```diff
@@ -291,9 +291,14 @@ exec::AggregateRegistrationResult registerArbitrary(const std::string& name) {
...
           case TypeKind::TIMESTAMP:
             return std::make_unique<ArbitraryAggregate<Timestamp>>(inputType);
+          case TypeKind::VARBINARY:
+            FOLLY_FALLTHROUGH;
           case TypeKind::VARCHAR:
+            FOLLY_FALLTHROUGH;
           case TypeKind::ARRAY:
+            FOLLY_FALLTHROUGH;
           case TypeKind::MAP:
+            FOLLY_FALLTHROUGH;
```
Comment (bikramSingh91) [inline velox/functions/prestosql/aggregates/ArbitraryAggregate.cpp:301]: nit: consider using [[fallthrough]] instead

**Review by bikramSingh91 — APPROVED** [review]

File: velox/expression/tests/ArgumentTypeFuzzer.cpp (hunk 2 above)
Comment (mbasmanova) [reply velox/expression/tests/ArgumentTypeFuzzer.cpp:56]: I started with maxDepth = 5, but Fuzzer was running very slowly. Hence, reduced to 2.

**Comment by mbasmanova** [comment]
Found another bug in min and max with complex inputs. These should check for null array elements, map values and struct fields and throw, but they don't.

* ARRAY comparison not supported for arrays with null elements
* ROW comparison not supported for fields with null elements
* MAP comparison not supported for null value elements

**Pull request closed without merging**